vsphere/internal/helper/network/network_helper.go

```diff
@@ -32,6 +33,59 @@ func FromPath(client *govmomi.Client, name string, dc *object.Datacenter) (objec
 	return finder.Network(ctx, name)
 }
 
+func FromNameAndDVSUuid(client *govmomi.Client, name string, dc *object.Datacenter, dvsUuid string) (object.NetworkReference, error) {
+
+	finder := find.NewFinder(client.Client, false)
+	if dc != nil {
+		finder.SetDatacenter(dc)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
+	defer cancel()
+	networks, err := finder.NetworkList(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	if len(networks) == 0 {
+		return nil, fmt.Errorf("%s %s not found", "Network", name)
+	}
+
+	if len(networks) == 1 && dvsUuid == "" {
+		return networks[0], nil
+	} else if len(networks) > 1 && dvsUuid == "" {
+		return nil, fmt.Errorf("path '%s' resolves to multiple %ss, Please specify", name, "network")
+
+		//handle cases with same port group names by checking the dv switch.
+	} else if dvsUuid != "" {
+
+		dvsObj, err := dvsFromUUID(client, dvsUuid)
+		if err != nil {
+			return nil, err
+		}
+		dvsMoid := dvsObj.Reference().Value
+
+		for _, network := range networks {
+			if network.Reference().Type == "DistributedVirtualPortgroup" {
+				dvPortGroup := object.NewDistributedVirtualPortgroup(client.Client, network.Reference())
+
+				var dvPortGroupObj mo.DistributedVirtualPortgroup
+
+				err = dvPortGroup.Properties(ctx, dvPortGroup.Reference(), []string{"config"}, &dvPortGroupObj)
+				if err != nil {
+					return nil, err
+				}
+
+				if dvPortGroupObj.Config.DistributedVirtualSwitch != nil &&
+					dvsMoid == dvPortGroupObj.Config.DistributedVirtualSwitch.Value {
+					return dvPortGroup, nil
```

Comment:
What if there are more than one DistributedVirtualPortgroups?

Reply:
Multiple DistributedVirtualPortgroups is fine, we will handle that case by checking the distributed virtual switch they belong to.